docs/root/configuration/http/http_filters/lua_filter.rst

```diff
@@ -244,6 +244,23 @@ data to send. *timeout* is an integer that specifies the call timeout in millise
 Returns *headers* which is a table of response headers. Returns *body* which is the string response
 body. May be nil if there is no body.
 
+httpCallAsync()
+^^^^^^^^^^^^^^^
+
+.. code-block:: lua
+
+  headers, body = handle:httpCallAsync(cluster, headers, body, timeout)
```

Comment:
drive-by comment; I know nothing about Lua but I would've expected, and would hope you would not return headers/body from an async function, but would instead have the user supply another optional argument, which would be some sort of closure or lambda to run when the async fetch was complete, passing the headers and body to that function.

Even better, I think 2 lambdas should be supplied: one that is passed the headers when available. The second is called with body-chunks, until there are no more. It would receive a bool param that would be 'true' when the request is complete.

Actually I haven't looked deeply at the Envoy infrastructure for HTTP Fetches, but modeling the Lua interface based on that pattern (which might differ from my suggestion above) would be even better.

The phrase 'fire and forget' doesn't seem good to me from a resource usage perspective.

Reply:
Ah, you're correct, this does not return headers/body. Pushed a change. Thanks for noticing that!

Reply:
Resource/memory wise, it looks like there are checks in the pipeline for this (I ran into it in previous pushes, and resolved them).

Reply:
@jmarantz and I talked about this offline. We will rename this `httpCallNonblocking` and proceed without callbacks.